core/controllers/collection_editor.py

```diff
@@ -196,6 +196,11 @@ def put(self, collection_id):
                     self.user_id, collection_id)
                 collection_services.index_collections_given_ids([
                     collection_id])
+            elif not is_public:
```

Comment:
This should probably be ``elif self.is_admin`` since we only want admins to be able to unpublish collections. Additionally, ``is_public`` is already guaranteed to be false here.